components/gitpod-cli/pkg/supervisor/terminal-attach.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package supervisor
+
+import (
+	"context"
+	"io"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/creack/pty"
+	"github.com/gitpod-io/gitpod/supervisor/api"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/term"
+)
+
+type AttachToTerminalOpts struct {
+	Interactive bool
+	ForceResize bool
+	Token       string
+}
+
+func AttachToTerminal(ctx context.Context, client api.TerminalServiceClient, alias string, opts AttachToTerminalOpts) {
```

Comment:
This is the same as https://github.com/gitpod-io/gitpod/blob/5934558596cebbfd179baa897b9126bec66a4c3c/components/supervisor/cmd/terminal-attach.go#L79

We agreed with @akosyakov to duplicate the logic instead of trying to share it between the two CLIs.

Reply:
Also, this function comes with a problem - if a terminal attaches to itself, it will go into a loop. In tasks cmd, we try avoiding entering in that situation but it's not easy to detect.

Reply:
Maybe, in the context of the running terminal, there could be an environment variable indicating it's a Task Terminal, so the attach command could check this env variable, and if detected someone trying run `gp tasks attach` on it, it would just print a message saying, for example: "It's not possible to attach to a task terminal from another task terminal. Please start a new terminal and run 'gp tasks attach' from there."